lib/utils/addon-utils.ts

```diff
@@ -8,4 +9,44 @@ import { ClusterAddOn } from '../spi';
  */
 export function getAddOnNameOrId(addOn: ClusterAddOn): string {
   return addOn.id ?? addOn.constructor.name;
+}
+
+/**
+ * Decorator function that accepts a list of AddOns and
+ * ensures addons are scheduled to be added as well as
+ * add them as dependencies
+ * @param addOns 
+ * @returns 
+ */
+export function dependable(addOns: Array<string>) {
```

Comment:
I suggest we use this form of declaration:

`dependable(...addOns: string )`

This will allow to declare dependency as  `@dependendable('AwsLoadbalancerController', 'SomeOtherAddon')`

Reply:
@askulkarni2 would love to see some tests for this.

Reply:
Yep, adding the tests.

Reply:
addressed